simplq/src/components/pages/Admin/AdminPage.jsx

```diff
@@ -16,8 +19,31 @@ import getToursteps from './getTourSteps';
 const TIMEOUT = 10000;
 let timeoutId;
 
-export default (props) => {
+const isQueueDeleted = (queueInfo) => queueInfo?.status === 'DELETED';
+
+const AdminPage = (props) => {
   const queueId = props.match.params.queueId;
+  const queueInfo = useSelector(selectQueueInfo);
+  const dispatch = useDispatch();
+  const history = useHistory();
+  const getQueueInfo = useCallback(useGetQueueInfo(), []);
+
+  useEffect(() => {
+    dispatch(getQueueInfo({ queueId }));
+  }, [dispatch, queueId, getQueueInfo]);
+
+  useEffect(() => {
+    if (queueInfo && queueId === queueInfo?.queueId && isQueueDeleted(queueInfo)) {
+      dispatch(setErrorPopupMessage('This queue is deleted.'));
+      history.push('/');
+    }
+  }, [dispatch, history, queueId, queueInfo]);
```

Comment:
We don't need this useEffect, I think. Can we have the if block directly over here?